### PR TITLE
Backport main - Metrics Explorer updated with PromQL (#1303)

### PR DESCRIPTION
**Pull request by pjfitzgibbons**
* Metrics Explorer updated with PromQL

including query for Prometheus Metrics and updated Export Panel

---------

Signed-off-by: Peter Fitzgibbons <peter.fitzgibbons@gmail.com>
(cherry picked from commit d16f2b2edfe0085295ac237bbfc9905c55210839)
Signed-off-by: Peter Fitzgibbons <peter.fitzgibbons@gmail.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .husky/.gitignore [hunk 1]
```diff
@@ -1 +0,0 @@
-_
```
Comment (ps48) [inline .husky/.gitignore:1]: Is the deletetion of husky files here intentional?

File: .husky/pre-commit [hunk 2]
```diff
@@ -1,4 +0,0 @@
-#!/bin/sh
-. "$(dirname "$0")/_/husky.sh"
-
-yarn lint-staged
```
Comment (ps48) [inline .husky/pre-commit:4]: Are we pushing this config to `package.json`?

File: public/components/application_analytics/__tests__/__snapshots__/trace_config.test.tsx.snap [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`Trace Config component renders empty trace config 1`] = `
-<TraceConfig
+<Component
```
Comment (ps48) [inline public/components/application_analytics/__tests__/__snapshots__/trace_config.test.tsx.snap:4]: I think the tests are failing here cause, the OSD is not on main branch.

**Review by kavithacm — APPROVED** [review]

File: .husky/.gitignore (hunk 1 above)
Comment (pjfitzgibbons) [reply .husky/.gitignore:1]: No.  I'll revert that.

File: .husky/pre-commit (hunk 2 above)
Comment (pjfitzgibbons) [reply .husky/pre-commit:4]: Reverted.

File: public/components/metrics/top_menu/__tests__/metrics_export.test.tsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    );
+
+    // Assert
+    const saveButton = await screen.findByText(/Save/);
+    expect(saveButton).toBeInTheDocument();
+    fireEvent.click(saveButton);
+
+    expect(await screen.findByText('Custom operational dashboards/application'));
+
+    userEvent.keyboard('{ESCAPE}');
```
Comment (ps48) [inline public/components/metrics/top_menu/__tests__/metrics_export.test.tsx:152]: Can replace this with `fireEvent.keyDown(element, { key: 'Escape', code: 'Escape' });`  As we don't have `'@testing-library/user-event'` installed.

**Comment by ps48** [comment]
I see now with snapshot updates all tests pass except one from metrics, due to package being missing. 

```
Summary of all failing tests
FAIL public/components/metrics/top_menu/__tests__/metrics_export.test.tsx
  ● Test suite failed to run

    Cannot find module '@testing-library/user-event' from 'public/components/metrics/top_menu/__tests__/metrics_export.test.tsx'

       6 | import React, { PropsWithChildren } from 'react';
       7 | import { render, screen, fireEvent } from '@testing-library/react';
    >  8 | import userEvent from '@testing-library/user-event';
         | ^
       9 | import type { RenderOptions } from '@testing-library/react';
      10 | import { configureStore } from '@reduxjs/toolkit';
      11 | import { Provider } from 'react-redux';

      at Resolver.resolveModule (../../node_modules/jest-resolve/build/resolver.js:324:11)
      at Object.<anonymous> (public/components/metrics/top_menu/__tests__/metrics_export.test.tsx:8:1)
      ```

File: public/components/metrics/top_menu/__tests__/metrics_export.test.tsx (hunk 4 above)
Comment (pjfitzgibbons) [reply public/components/metrics/top_menu/__tests__/metrics_export.test.tsx]: Done.

**Comment by pjfitzgibbons** [comment]
> I see now with snapshot updates all tests pass except one from metrics, due to missing package.

Reverted

**Review by ps48 — APPROVED** [review]
Ignoring the linting errors for now, as linter is not backported to 2.x and this PR is already merged in 2.x

**Pull request merged**